oak-search-elastic/src/main/java/org/apache/jackrabbit/oak/plugins/index/elastic/query/ElasticIndex.java

```diff
@@ -96,16 +96,16 @@ protected IndexNode acquireIndexNode(String indexPath) {
     }
 
     @Override
-    protected String getFulltextRequestString(IndexPlan plan, IndexNode indexNode) {
-        return Strings.toString(new ElasticRequestHandler(plan, getPlanResult(plan)).baseQuery());
+    protected String getFulltextRequestString(IndexPlan plan, IndexNode indexNode, NodeState rootState) {
+        return Strings.toString(new ElasticRequestHandler(plan, getPlanResult(plan), rootState).baseQuery());
     }
 
     @Override
     public Cursor query(IndexPlan plan, NodeState rootState) {
         final Filter filter = plan.getFilter();
         final FulltextIndexPlanner.PlanResult planResult = getPlanResult(plan);
 
-        final ElasticRequestHandler requestHandler = new ElasticRequestHandler(plan, planResult);
+        final ElasticRequestHandler requestHandler = new ElasticRequestHandler(plan, planResult, rootState);
```

Comment:
I assume this rootState is to get the vector values from the image node since the similarity query would only have the path to the image node.

If we want to avoid changing this signature  - what we can do is call ES twice - first time get the node data for the path in query, read the value for vector dims and then use them in the similarity query

Reply:
We discussed this earlier in scrum and concluded that we should avoid unnecessary remote calls.

Reply:
ok